src/modules/joystick/Joystick.h

```diff
@@ -144,6 +144,26 @@ class Joystick : public Object
 		INPUT_TYPE_MAX_ENUM
 	};
 
+	enum PowerState
+	{
+		POWERSTATE_ERROR = -1,   /**< error determining power status */
+		POWERSTATE_UNKNOWN,      /**< cannot determine power status */
+		POWERSTATE_ON_BATTERY,   /**< Not plugged in, running on the battery */
+		POWERSTATE_NO_BATTERY,   /**< Plugged in, no battery available */
+		POWERSTATE_CHARGING,     /**< Plugged in, charging battery */
+		POWERSTATE_CHARGED,       /**< Plugged in, battery charged */
+		POWERSTATE_MAX_ENUM
+	};
+
+	enum ConnectionState
```

Comment:
This (and the related methods/functions) could be renamed to `ConnectionType` for consistency with other APIs.